extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSearchExtractor.java

```diff
@@ -103,135 +132,109 @@ public boolean isCorrectedSearch() {
     public List<MetaInfo> getMetaInfo() throws ParsingException {
         return YoutubeParsingHelper.getMetaInfo(
                 initialData.getObject("contents").getObject("twoColumnSearchResultsRenderer")
-                        .getObject("primaryContents").getObject("sectionListRenderer").getArray("contents"));
+                        .getObject("primaryContents").getObject("sectionListRenderer")
+                        .getArray("contents"));
     }
 
     @Nonnull
     @Override
     public InfoItemsPage<InfoItem> getInitialPage() throws IOException, ExtractionException {
         final InfoItemsSearchCollector collector = new InfoItemsSearchCollector(getServiceId());
 
-        final JsonArray sections = initialData.getObject("contents").getObject("twoColumnSearchResultsRenderer")
-                .getObject("primaryContents").getObject("sectionListRenderer").getArray("contents");
+        final JsonArray sections = initialData.getObject("contents")
+                .getObject("twoColumnSearchResultsRenderer").getObject("primaryContents")
+                .getObject("sectionListRenderer").getArray("contents");
 
         Page nextPage = null;
 
         for (final Object section : sections) {
             if (((JsonObject) section).has("itemSectionRenderer")) {
-                final JsonObject itemSectionRenderer = ((JsonObject) section).getObject("itemSectionRenderer");
+                final JsonObject itemSectionRenderer = ((JsonObject) section)
+                        .getObject("itemSectionRenderer");
 
                 collectStreamsFrom(collector, itemSectionRenderer.getArray("contents"));
-
-                nextPage = getNextPageFrom(itemSectionRenderer.getArray("continuations"));
```

Comment:
There is always a next page, see line 157. I removed the old search code because it isn't used anymore with the new API: I renamed `getNewNextPageFrom` to `getNextPageFrom`.